src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -461,15 +463,33 @@ async def resample(self, *, one_shot: bool = False) -> None:
                 timeseries from the resampler before calling this method
                 again).
         """
-        while True:
-            await self._wait_for_next_resampling_period()
+        # We use a tolerance of 10% of the resampling period
+        tolerance = timedelta(
+            seconds=self._config.resampling_period.total_seconds() / 10.0
+        )
+        async for drift in Timer.periodic(
+            timedelta(seconds=self._config.resampling_period.total_seconds())
+        ):
```

Comment:
This timer needs to be a instance member, otherwise the `drift` will always be zero, as you creating the timer on each resample call. For the Resampling actor this is not an issue because `resample()` is called once and then we loop on it, but in the tests we use `one_shot` and in that case we call `resample()` multiple times, thus triggering the permanent zero drift.

Also the tests need to be adjusted to also do a `asyncio.sleep()` with the same time as `fake_time.shift()` to make sure the loop monotonic clock (which is the one the new channels timer uses) is also advanced to produce a drift, otherwise the timer will always trigger at the interval time.